### ISSUE TITLE
add thread-safe monitors to c-api branch

### ISSUE THREAD

**Issue by ingowald** [owl-project/owl#4]

**Comment by ingowald** [comment]
maybe better solution: add to all methods of devicegroup instead of dll entry points.

**Comment by ingowald** [comment]
all variables and context handles are protected by mutexes; and can be created/read by many threads in parallel. builds etc may not yet be thread safe, but should be tracked in another issue.
